app/validation/validator.py

```diff
@@ -1,6 +1,98 @@
+from app.validation.validation_result import ValidationResult
+from app.validation.required import Required
+from app.validation.type_validator_factory import TypeValidatorFactory
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+class ValidationException(Exception):
+    pass
+
+
```

Comment:
Haven't renamed it, but I have no strong convictions on this either way.  When this component was originally discussed we used the terms Validator and ValidationEngine interchangeably.